.github/pull_request_template.md

```diff
@@ -1,20 +1,24 @@
-<!--
-
-
-  (ﾉ◕ヮ◕)ﾉ*:・ﾟ✧
-  
-  Thank you for your interest in contributing to the k6 project!
-  
-  Before you get started, we'd kindly like to ask you to read our:
-    - Contribution guidelines at https://github.com/grafana/k6/blob/master/CONTRIBUTING.md
-    - Code of Conduct at https://github.com/grafana/k6/blob/master/CODE_OF_CONDUCT.md
-    
-  Out of respect for your time, please start a discussion regarding any bigger contributions either
-  in a GitHub Issue, in the community forums or in the #contributors channel of the k6 slack before you
-  get started on the implementation.
-  
-  If you've already done all of that, you're more than welcome to proceed with your pull request.
-  Thank you again for your contribution! 🙏🏼
-  
-  
--->
+## What?
+
+<!-- A short (or detailed) description of what this PR does. -->
+
+## Why?
+
+<!-- A short (or detailed) explanation of why these changes are made. -->
+
+## Checklist
+
+<!-- Only answer those that apply to your PR: -->
+
+- [ ] I have performed a self-review of my code.
+- [ ] I have added tests for my changes.
+- [ ] I have run linter locally (`make ci-like-lint`) and all checks pass.
+- [ ] I have run tests locally (`make test`) and all tests pass.
+- [ ] I have commented on my code, particularly in hard-to-understand areas.
+- [ ] I have read the [contribution guidelines](https://github.com/grafana/k6/blob/master/CONTRIBUTING.md).
+- [ ] I have read the [code of conduct](https://github.com/grafana/k6/blob/master/CODE_OF_CONDUCT.md).
```

Comment:
Sorry, I find checklists like these pointless at best, and annoying at worst.

The first 5 are general processes any developer should be familiar with, and we don't need to request them explicitly. If they're not done, that means that they haven't read our contribution guidelines, which should, and mostly does, mention these. Even then, 2 of those are automated checks that would just fail in CI, and they would get the hint that way.

Since these don't apply to all PRs, in some cases we will have an empty list of things that just clutter the description.

Why does a contributor need to confirm they've read our contribution guidelines and CoC on every PR? That happens once, and then they wouldn't need to check the points above, assuming everything is mentioned in the guidelines.

We already link to the contribution guidelines and CoC from the commented out description, which I think is good enough. Do we spend so much time in PRs discussing these specific points that it's worth cluttering all PR descriptions from now on?

So I'm very much against this, but if the rest of the team thinks it's a good thing, then I'll live with it. :man_shrugging:

Reply:
I'm interpreting it as we use this mostly for external contributors.  I expect us to use the first two questions and then we delete the checklist. If this assumption is wrong then I'm also against it.

>**and mostly does**

I think @olegbespalov has a different feeling if it is opening this PR.

>then, 2 of those are automated checks that would just fail in CI, and they would get the hint that way.

The CI is a slow process because we have to approve the run on every new commit. This is the main helpful point for me of this template. You may read the Guidelines but after you made some changes you forgot to run the linter again. Personally, I've seen it several times in external contributors' PRs.

Reply:
I’d agree with @imiric on the last two ones. While I agree it’s good to mention the guidelines and CoC, I don’t think a checkbox there would help anything besides annoying contributors. How about we keep these statements, but remove checkboxes for them?

Reply:
> Even if they should, it doesn't mean that they will. Also, remember that we have different contributors with different backgrounds, so it's always good to ask the second time.

But that's what we have the contribution guidelines for. If they haven't read those and done the things we suggest, why would ticking a checkbox suddenly make them? If these are not mentioned in the guidelines, then we should update them.

In practice, checklists won't improve the quality of the PRs we get, and will only annoy contributors and reviewers. But you guys clearly like them, so majority rules. :slightly_smiling_face: